Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -285,21 +281,25 @@ namespace AzToolsFramework::ComponentModeFramework
     {
         if (mode == ViewportEditorMode::Component)
         {
-            // when component mode ends, change the active button to transform
             m_activeSwitcherComponent = nullptr;
 
-            bool inComponentMode = false;
-            AzToolsFramework::ComponentModeFramework::ComponentModeSystemRequestBus::BroadcastResult(
-                inComponentMode, &ComponentModeSystemRequests::InComponentMode);
+            // wait one frame to check if component mode has been re-entered before changing switcher button to transform
+            AZ::TickBus::QueueFunction(
+                [this]()
+                {
+                    bool inComponentMode = false;
+                    AzToolsFramework::ComponentModeFramework::ComponentModeSystemRequestBus::BroadcastResult(
+                        inComponentMode, &ComponentModeSystemRequests::InComponentMode);
 
-            if (!inComponentMode)
-            {
-                ViewportUi::ViewportUiRequestBus::Event(
-                    ViewportUi::DefaultViewportId,
-                    &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
-                    m_switcherId,
-                    m_transformButtonId);
-            }
+                    if (!inComponentMode)
+                    {
+                        ViewportUi::ViewportUiRequestBus::Event(
+                            ViewportUi::DefaultViewportId,
+                            &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
+                            m_switcherId,
+                            m_transformButtonId);
+                    }
+                });
```

Comment:
The edge case would be to edit and save a large texture, like 8192x8192, since this is guaranteed to bring up a progress bar, which will cause several ticks to run. It could be that the previous changes to the Switcher where we changed up the shutdown notification ordering already fixed any issues here. Thanks for verifying this!